### ISSUE TITLE
How to modify the evaluation metrics for hyperparameter tuning?

### ISSUE THREAD

**Issue by jsJSjs666** [ultralytics/ultralytics#18726]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

The default code seems to be mAP50-95, how to change to F1 score?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jsJSjs666, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question or a functionality-related inquiry (like modifying evaluation metrics), please provide as much information as possible, including details about your setup, such as the version of the `ultralytics` package you’re using, any custom configs, and relevant logs if applicable. Verifying that you’ve followed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) will also help.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure you are using the latest version featuring the most up-to-date functionality and bug fixes:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these supported, up-to-date environments (with dependencies preinstalled):

- **Notebooks** with free GPU access: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Your question regarding changing evaluation metrics in the code for hyperparameter tuning is noted. This response is automated 🤖, but an Ultralytics engineer will respond soon to help further! 🎉

**Comment by Y-T-G** [comment]
Are you using Ray tune or normal tuning?

**Comment by jsJSjs666** [comment]
> Are you using Ray tune or normal tuning?
from ultralytics import YOLO

# Initialize the YOLO model
model = YOLO(r'F:\NJU307_js\ultralytics-main-Acconv-3\ultralytics\cfg\models\v8\myyolov8.yaml')

# Tune hyperparameters on COCO8 for 30 epochs
model.tune(
    data=r'F:\ultralytics-main-0\ultralytics\cfg\datasets\mycoco128.yaml',
    epochs=30,
    iterations=100,
    optimizer="AdamW",
    plots=True,
    save=True,
    val=True,
    device='1',
)

I use this code to run hyperparameter tuning.

**Comment by Y-T-G** [comment]
You can change the weights and make F1-score 1 and rest 0

https://github.com/ultralytics/ultralytics/blob/b1042c4da81c7c5da6a825b92f22610c58f58ea1/ultralytics/utils/metrics.py#L750
